applications/FluidDynamicsApplication/custom_processes/embedded_postprocess_process.h

```diff
@@ -0,0 +1,247 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+//
+
+#ifndef KRATOS_EMBEDDED_POSTPROCESS_PROCESS_H
+#define KRATOS_EMBEDDED_POSTPROCESS_PROCESS_H
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "processes/process.h"
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "includes/cfd_variables.h"
+#include "utilities/openmp_utils.h"
+
+// Application includes
+
+
+namespace Kratos
+{
+///@addtogroup FluidDynamicsApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// Utility to filter the embedded velocity and pressure values
+class EmbeddedPostprocessProcess : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of EmbeddedPostprocessProcess
+    KRATOS_CLASS_POINTER_DEFINITION(EmbeddedPostprocessProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+    EmbeddedPostprocessProcess(ModelPart& rModelPart)
+    {
+        mrModelPart = rModelPart;
+    }
+
+    /// Destructor.
+    virtual ~EmbeddedPostprocessProcess(){}
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    void ExecuteFinalizeSolutionStep() override
+    {
+        const array_1d<double, 3> aux_zero = ZeroVector(3);
+        ModelPart::NodesContainerType& rNodes = mrModelPart.Nodes();
+
```

Comment:
verify here that the first node has the variables you need, and eventually throw an error.